frontend/src/bundles/common/components/progress/progress.tsx

```diff
@@ -0,0 +1,29 @@
+import styles from './styles.module.scss';
+
+type Properties = {
+    fillPercent: number;
+};
+
+const Progress: React.FC<Properties> = ({ fillPercent }) => {
+    // sets valid fill percent in range between 0 and 100
+    const validFillPercent: number = Math.min(100, Math.max(0, fillPercent));
+    // removes right side border radius when a fill percent lower than 95%
+    const innerFillBorderRadius: string =
```

Comment:
It'd better to find other solution, without any calculations and magic numbers (95)